### PR TITLE
Selection component pop

### PR DESCRIPTION
**Pull request by Arivias**
## About the PR 
Adds a "remove current + send" input to the selection component.

## Why's this needed? 
Makes it more convenient to use the selection component as a queue.

### CODE REVIEW

**Comment by Arivias** [comment]
Not quite sure what happened with line 999 there...

**Review by ZeWaka — APPROVED** [review]
looks dank

**Comment by ZeWaka** [comment]
> Not quite sure what happened with line 999 there...

Caused by the editor removing errant spacing on that line. Intended behavior.

File: code/WorkInProgress/MechanicMadness.dm [hunk 1]
```diff
@@ -2007,6 +2008,19 @@
...
+		if(level == 2 || !input) return
+		if(current_index <= length(signals))
+			LIGHT_UP_HOUSING
+			sendCurrent(input)
+			signals.Remove(signals[current_index])
+			if(current_index > length(signals))
+				current_index = length(signals) ? length(signals) : 1 // Don't let current_index be 0
+			tooltip_rebuild = 1
+			if(announce)
+				componentSay("Removed : [input.signal]")
```
Comment (MarkNais) [inline code/WorkInProgress/MechanicMadness.dm:2021]:
```suggestion
		sendCurrent(input)
		remitem(input)
```
All the same checks and effects are triggered in these two functions.
`sendCurrent()` also modifies the signal `input`, so the correct item will be removed when passed to `remitem()`

Comment (MarkNais) [inline code/WorkInProgress/MechanicMadness.dm:2021]:
```suggestion
		sendCurrent(input)
		remitem(input)
```
All the same checks and effects are triggered in these two functions.
`sendCurrent()` also modifies the signal `input`, so the correct item will be removed when passed to `remitem()`
Also I tested this on local and it worked well.

**Review by MarkNais — CHANGES_REQUESTED** [review]

**Review by MarkNais — APPROVED** [review]

File: code/WorkInProgress/MechanicMadness.dm [hunk 2]
```diff
@@ -1897,6 +1897,7 @@
 		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_INPUT,"add item", "additem")
 		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_INPUT,"remove item", "remitem")
 		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_INPUT,"remove all items", "remallitem")
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_INPUT,"remove current + send", "popitem")
```
Comment (ZeWaka) [inline code/WorkInProgress/MechanicMadness.dm:1900]: Shouldn't this be named "send current + remove" or "pop current"?

Comment (Arivias) [reply code/WorkInProgress/MechanicMadness.dm]: yeah, it could be worded more clearly.

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**